### ISSUE TITLE
Feature request: Select extruder/tool for M571 (Set output on extrude)

### ISSUE THREAD

**Issue by timmit99** [dc42/RepRapFirmware#80]
I was hoping there could be a parameter (D) that would allow you to select which extruder/tool you want to trigger the output on extrude. That way you can have the output go high only when a selected tool is operational.

**Comment by dc42** [comment]
Can you tell me that application that you have in mind?

You could use M571 commands in the tool change files to enable/disable output depending on which tool is selected, but I appreciate that you might want something more secure if you are controlling a laser.

**Comment by timmit99** [comment]
yes, i have a combo laser cutter/mill and i sometimes do 2 operations in a single file. My first "extruder" (tool) is the laser cutter and the second "extruder" (tool) is a spindle. I will etch the plate and then mill the outer edges to shape, being able to have the laser on ONLY when i'm using the laser section would be nice, as of now i'm manually going through the code and triggering it on and off with a python script. If i could assign the "Trigger on extruder" just for the laser that would be ideal.

**Comment by T3P3** [comment]
this would also be useful for an academic project.
